### ISSUE TITLE
Parking Space Permit Fee Insertion (On PROD🎉🎉)

### ISSUE THREAD

**Issue by vschandramouli** [cityofaustin/atd-data-tech#7912]
**Acceptance Criteria:**
- System should insert the Parking Space Permit Fee (50170) when 'Approved' attempt is made in Parking Parking Review Process (50662) in RW folder of subtype Parking Spaces and worktype Load/Unload 
- System should also create a bill when the Parking Space Permit Fee (50170) is inserted in Right of Way folder of subtype Parking Spaces (Fee code exists, logic needs to be updated)
- The amount for Parking Space Permit Fee should be calculated as follows:
Metered Fee = Value in Info Rate if metered (in $ per parking meter)   **X**   Value in 'Number of Days' Info field (Info Code 75390)  **X**   16   **X**   Number of 'Parking Spaces : Meter Number' Rows in Property & Street Information Freeform 
Unmetered Fee = $3.20   **X**   Value in 'Number of Days' Info field (Info Code 75390)   **X**   Value in 'Number of (#)' field in 'Parking Spaces : # Unmetered Spaces' row of Property & Street Information Freeform
Parking Space Permit Fee (50170) = Metered Fee + Unmetered Fee
- After UAT, user asked to add ATD Tech surcharge 4% for the added Parking Space Permit Fee. (added by Subha)

**Note** 
- '**X**' in the payment logic above denotes multiplication
- Values 16 and $3.20 are variables and may change in the future. These should not be hard coded and should be added to a lookup table (Tracy mentioned this was lookup table 5025)
- The logic for inserting the fee may currently exist. If it does, it needs to be updated or removed to account for this new change in logic
- A screenshot of Property & Street Information Freeform provided below for reference
![image.png](https://images.zenhubusercontent.com/5f50fe1f66d1a40f7f557fb4/4a74ca42-471a-4264-89fe-1ad733410534)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  @vschandramouli  

- I didn't find 5025 lookup table, 50200 is the lookup table for 50170 fee.
- As per the existing code, 50170 fee is also applicable for RW TURP. So if we change the logic, that is going to be applicable for RW TURP(PVT, PCO). Is it going to change for TURP as well?
- If the fee calculation is different in both subtypes, we may need to add new lookup table entries Parking Space subtype and write new code logic.
Kindly, advise to proceed further.

**Comment by Nadin-Nader** [comment]
@SubhaShroff i asked Tracy. Im just waiting on a response from her or to see if i have to ask Paloma.

**Comment by TracyLinder** [comment]
@Nadin-Nader please reach out to @vschandramouli for clarification. He's the SPOC on this one. Thanks!

**Comment by TracyLinder** [comment]
@SubhaShroff @Nadin-Nader  @vschandramouli 

If I understand correctly, fee 50170 Parking Space Permit Fee is also coded to the TURP subtype. If we change it on parking it will also change on TURP. 

My feedback is:
- Preferably we'd have a unique fee code for the parking space permit fee that isn't used on any other permit types. 
- Do ROW users ever insert this fee on TURP permits? (It's not a TURP fee.) If not, can we remove this fee from being connected to the TURP subtype?

**Comment by Nadin-Nader** [comment]
@SubhaShroff here is Paloma's response:

[1:32 PM] Amayo-Ryan, PalomaI'll give it a shot:


I didn't find 5025 lookup table, 50200 is the lookup table for 50170 fee. 
_**I'm not sure what a lookup table is exactly ... or where those codes came from BUT I can confirm that the info code for the Parking Space Permit Fee is 50170.**_
As per the existing code, 50170 fee is also applicable for RW TURP. So if we change the logic, that is going to be applicable for RW TURP(PVT, PCO). Is it going to change for TURP as well?  
_**We should not be using 50170 on the TURP subtype.  I am not sure why it is currently showing as applicable or how changes to it will impact the TURP subtype.  The relevant fee used on TURPs is "TURP Metered Parking Space" info code 50321.  There should be no changes to "TURP Metered Parking Space" 50321.**_ 
For this one Tracy added: Preferably we'd have a unique fee code for the parking space permit fee that isn't used on any other permit types.  
_**Agreed.  And I think/thought we already do.**_
If the fee calculation is different in both subtypes, we may need to add new lookup table entries Parking Space subtype and write new code logic.  
**_The fee calculation is the same but it should be two different fees. Kindly, advise to proceed further._** 
Also had a followup question:
Do ROW users ever insert this fee on TURP permits? (It's not a TURP fee.) If not, can we remove this fee from being connected to the TURP subtype?  
**_ROW users should not insert fee 50170 on a TURP.  I'm fine with removing the fee but I am concerned something will break (or at least that seems to happen often)._**

**Comment by Nadin-Nader** [comment]
@SubhaShroff paloma had a followup question:

[1:34 PM] Amayo-Ryan, Paloma

Nader, Nadin Actually ... I have one question: When was fee 50321 added?  I'm suspecting that at one point we didn't have 50321 so we used 50170 instead.  Depending on when 50321 was added that may mean that we have some older, still Active, TURPs that are relying on 50170.

**Comment by Nadin-Nader** [comment]
Waiting to continue the work when Jonathan comes back on Wednesday 3/2/22 to give us clarification on the logic and formula.

**Comment by Nadin-Nader** [comment]
Subha's convo with Paloma:

> [Yesterday 4:19 PM] Amayo-Ryan, PalomaTicket 7912
    
Nader, Nadin and Linder, Tracy - I spoke with Jonathan Wheeler regarding this ticket.  He suggested one rephrasing: Instead of saying "$ per parking meter" it would be more accurate to say "hourly meter rate".  If anyone has any follow up questions he's back in the office and happy to work directly with whoever has questions!




​[Yesterday 4:38 PM] Shroff, Subhashini - Vendor
    Amayo-Ryan, Paloma "hourly meter rate" is it another folder info? may I know the infocode for that? And also I had question about the variables "16" & "$3.20"
​[Yesterday 4:44 PM] Amayo-Ryan, Paloma
    It is not another folder info.  It would be a more accurate name for the new folder info that was recently added.

Both 16 and $3.20 are correct variables.
16 indicates the total number of serviceable hours per metered space per day.
$3.20 equals the daily rater per unmetered space per day.
​[Yesterday 4:46 PM] Shroff, Subhashini - Vendor
    Thank you.
<https://teams.microsoft.com/l/message/19:6a6af073328c4ad39cfa3c7432b91629@thread.tacv2/1646259586426?tenantId=5c5e19f6-a6ab-4b45-b1d0-be4608a9a67f&amp;groupId=ab15580b-43a7-4982-a971-ba8e9f1426fc&amp;parentMessageId=1646259586426&amp;teamName=ROW - AMANDA Support&amp;channelName=Testing Feedback&amp;createdTime=1646259586426>

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  @vschandramouli 
I have sent a demo for this ticket in email. Please review.

**Comment by Nadin-Nader** [comment]
@TracyLinder  @SubhaShroff Demo looks good to go to **DEV**

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0205698 is raised for DEV deployment.

**Comment by Nadin-Nader** [comment]
@SubhaShroffThis has been deployed to DEV, Please UAT and let me know the outcome

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0206186 is submitted for DEV deployment of commits - 
"c3df7a2261fe" 
"91878deb92e8"

**Comment by SubhaShroff** [comment]
@Nadin-Nader  I tested this ticket on DEV. Looks good, please let me know if i can move this TEST.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  TEST deployment request is submitted - SCTASK0206469.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  I tested it on TEST, works as expected in AC. Ready for UAT.

**Comment by Nadin-Nader** [comment]
[Wednesday 5:37 PM] Amayo-Ryan, PalomaTest Results for Ticket 7912
    
Nader, Nadin and Linder, Tracy
[4:00 PM] Wheeler, Jonathan

Upon testing, it does put in the correct fees when you put 2 in the metered rate. However, it creates a bill but does not include the 4% surcharge. Is there a way to have it to where it creates the bill with both the fee and the surcharge or make it to where it doesn't bill the invoice but starts the invoice (i.e. does not assign a bill number until you manually hit create)? If not, this will involve manually creating a new invoice for the surcharge and then sending the applicant two invoices instead of one.

**Comment by SubhaShroff** [comment]
@Nadin-Nader [SCTASK0207352 is submitted for DEV deployment request to add the surcharge.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The DEV deployment is done and I tested it for surcharge addition. Works as expected. Please let me know if I can submit a TEST deployment request for this.

**Comment by SubhaShroff** [comment]
### Status:
SCTASK0207789 is submitted for TEST deployment.

**Comment by Nadin-Nader** [comment]
This has been sent to the DEVOPS CM Team to be deployed to **_PROD_** Under **SCTASK0208760**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  this ticket is deployed in PROD.

**Comment by SubhaShroff** [comment]
### Status:
Almost 13days ago, it was deployed on PROD.

**Comment by TracyLinder** [comment]
This is OK to close.

**Comment by TracyLinder** [comment]
### Status

- Paloma was not aware this went to PROD. We need to post it in the teams chat & leave this open for another week.

**Comment by Nadin-Nader** [comment]
> [Yesterday 7:02 PM] Amayo-Ryan, Paloma
>     Nader, Nadin for 8108 BUG, I believe Linder, Tracy tested that during our Friday meeting and confirmed that it was all systems go.  I want to say that she tested another card as well but I can't remember now which one.
> 
> For 7912 everything appears to be working fine so I'm a-okay with closing it out.

Closing card
